drf_braces/tests/test_utils.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import print_function, unicode_literals
+
+import unittest
+
+from drf_braces.utils import (
+    get_class_name_with_new_suffix,
+    get_attr_from_base_classes,
+)
+from rest_framework import fields
+
+
+class TestUtils(unittest.TestCase):
+    def test_get_class_name_with_new_suffix(self):
+        new_name = get_class_name_with_new_suffix(
+            klass=fields.IntegerField,
+            existing_suffix='Field',
+            new_suffix='StrawberryFields'
+        )
+        self.assertEqual(new_name, 'IntegerStrawberryFields')
+
+        new_name = get_class_name_with_new_suffix(
+            klass=fields.IntegerField,
+            existing_suffix='straws',
+            new_suffix='Blueberries'
+        )
+        self.assertEqual(new_name, 'IntegerFieldBlueberries')
+
+    def test_get_attr_from_base_classes(self):
+        Parent = type(str('Parent'), (), {'fields': 'pancakes'})
+
+        self.assertEqual(
+            get_attr_from_base_classes((Parent,), [], 'fields'), 'pancakes'
+        )
+
+        self.assertEqual(
+            get_attr_from_base_classes(
+                (Parent,), {'fields': 'mushrooms'}, 'fields'
+            ),
+            'mushrooms'
+        )
+
+        self.assertEqual(
+            get_attr_from_base_classes((Parent,), [], '', default='maple_syrup'),
```

Comment:
this does not return str or default. this returns value found in either using these priorities:

1. dict of attributes (passed in `__new__`)
2. in parent bases

this is really meant to mimick:

```python
class A(object):
    foo = 1
class B(B):
    foo = 2

assert type('C', (B,) {'foo': 3}).foo == 3
assert type('C', (B,) {}).foo == 2
assert type('C', (A,) {}).foo == 1
```

so in https://github.com/agamdua/django-rest-framework-braces/blob/test/utils/drf_braces/serializers/form_serializer.py#L120-L122 I lookup if `_options_class` is defined as explicit attribute in the to-be created class. if yes, I use that. if not, I try to find it in any of the bases. and if not found there, I return default which uses `FormSerializerOptions`